homeassistant/components/heos/media_player.py

```diff
@@ -161,9 +160,7 @@ async def async_added_to_hass(self) -> None:
             async_dispatcher_connect(self.hass, SIGNAL_HEOS_UPDATED, self._heos_updated)
         )
         # Register this player's entity_id so it can be resolved by the group manager
-        self.hass.data[HEOS_DOMAIN][DATA_ENTITY_ID_MAP][self._player.player_id] = (
-            self.entity_id
-        )
+        self._group_manager.entity_id_map[self._player.player_id] = self.entity_id
```

Comment:
Existing bug, but couple of things:

1. Signaling that a player was added, maybe that can be done inside group manager by making registering a player a method, instead of directly mutating the dict.
2. You need to remove the entity from `group_manager.entity_id_map` when the entity is removed (you can consider returning a unregister function from the method of point 1, and then pass that to `self.async_on_remove`.

Reply:
Yeah, I noticed this too. I'll address it in a future PR. Since the group manager is scoped to the config entry (and only 1 instance is supported), I can't think of a scenario where an entity would get removed (I don't think the HEOS API has an event for a device going away.) However, I'll still make sure the code handles it, in case it does in the future. :)